orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTask.kt

```diff
@@ -77,6 +80,10 @@ constructor(
       TaskResult.builder(ExecutionStatus.SUCCEEDED).context(emptyMap<String, Any?>()).build()
     } catch (e: RetrofitError) {
       handleRetryableFailures(e, context)
+    } catch (networkException: SpinnakerNetworkException) {
```

Comment:
What about SpinnakerConversionException?  Why not catch SpinnakerServerException here and write one new handleRetryableFailures method that takes SpinnakerServerException?

Reply:
We definitely need test coverage for whatever new method(s) we end up with here.

Reply:
Added a catch block to handle the conversion errors. With this change, a behaviour change is expected . 
To demonstrate the same, I have made a specific commit on it - 099b09ea78bbc23fbf97f586747e406105995b40

RetrofitError.conversionError contains status code in it, whereas the SpinnakerConversionException does not consist this, and hence a behaviour change is expected

Reply:
Conversion errors only happen on 2xx responses, so leaving it out doesn't seem so bad.  It does also seem straightforward to add a response code member to SpinnakerConversionException.  We could even add a HasResponseCode interface that both SpinnakerConversionException and SpinnakerHttpException implement if that helps.

Reply:
Removed the catch block to handle the SpinnakerConversionException. As the return type of the API : keelService.publishDeliveryConfig is retrofit.client.Response. The conversion error never happens